### PR TITLE
Issue66 - Improved Error Handling and Unit Tests for check_panel.py

### LINKED ISSUE #66: check_panel.py : improve logging and add unit testing
Add unit testing to check_panel.py and improve logging

### PR DESCRIPTION
**Pull request by ASendellPrice**
This pull request replaces #72. 

This PR resolves #66, introducing several improvements to the check_panel.py script, aimed at enhancing its robustness and testability:

**Error Handling Improvements:**
- Enhanced error handling across the script to better capture and log issues that may arise during API requests or data processing.
- Error logging is now fully integrated with the centralised logging configuration system developed as part of Spring3 (see #71), ensuring consistency across modules.

**Unit Tests:**
- A full suite of unit tests has been developed for the key functions in check_panel.py, ensuring correct behaviour and error handling.
- The tests cover different scenarios and edge cases, providing better coverage and confidence in the script's reliability.
- All unit tests passed
image

**Preparation for build:**
- main.py script has been developed to handle the entry point for the application.
- setup.py has been added to facilitate package installation and configuration.

**Important Notes:**
- File Relocation: The settings.py file, which contains the logging configuration, will need to be moved to the PanelPal directory for the build to function correctly.

**PEP8:**
- check_panel.py: 
image

- main.py:
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,48 +23,73 @@ To set up a conda environment for this project, you can use the provided `enviro
...
+    ```bash
+    cd PanelPal
+    pip install .
+    ```
 
-Once a valid panel ID is entered, the tool will fetch information from the PanelApp API and display details about the selected panel.
+## Usage
 
-You will be asked to confirm your selection before proceeding. If confirmed, the tool will generate a BED file with gene locus information.
+### check panel
```
Comment (PatrickWeller) [inline README.md:49]: Maybe make this as "Check Panel" given the code below explains proper usage with lower case. The upper just looks nicer. Do the same for "generate bed file" below

File: test/test_check_panel.py
Comment (PatrickWeller) [inline test/test_check_panel.py:1]: Tests could have increased coverage. Please see image below.
image

File: test/test_panelpal.py
Comment (PatrickWeller) [inline test/test_panelpal.py:1]: Please remove file and commit the removal.

**Review by PatrickWeller — CHANGES_REQUESTED** [review]
Hi Ash,

Please see comments on files for your review before I merge with development.

File: README.md [hunk 2]
```diff
@@ -23,48 +23,73 @@ To set up a conda environment for this project, you can use the provided `enviro
...
+```
 
-Panel ID: R59
-Clinical indication: Hereditary Cancer Panel
+### generate bed file
+To generate a bed file for a given panel:
 
-Is this correct? (yes/no): yes
+```bash
+python PanelPal/check_panel.py --panel_id R207 --panel_version 4 --genome_build GRCh38
```
[the remark this answers is not in the document]
Comment (ASendellPrice) [reply README.md]: Corrected to 'python PanelPal/generate_bed.py --panel_id R207 --panel_version 4 --genome_build GRCh38'
See commit [e6bc2b5](https://github.com/PatrickWeller/PanelPal/pull/74/commits/e6bc2b5a274e5a8801a56f6b46c11d07b78262b4)

File: README.md [hunk 3]
```diff
@@ -23,48 +23,73 @@ To set up a conda environment for this project, you can use the provided `enviro
...
+```
 
-Proceeding with the selected panel.
+## Directory structure
+The following structure should be used going foward to keep the project directories tidy and in preperation for package build. This will also resolve issues importing modules going forward. Note: DB directory has been ommitted from the tree for now.
```
[the remark this answers is not in the document]
Comment (ASendellPrice) [reply README.md]: Typos in README.md corrected. See commit [e6bc2b5](https://github.com/PatrickWeller/PanelPal/pull/74/commits/e6bc2b5a274e5a8801a56f6b46c11d07b78262b4)

File: env/environment.yaml
[the remark this answers is not in the document]
Comment (ASendellPrice) [reply env/environment.yaml:1]: environment.yaml updated. Channels now in the following order:
- conda-forge
- defaults
- bioconda

see commit [e6bc2b5](https://github.com/PatrickWeller/PanelPal/pull/74/commits/e6bc2b5a274e5a8801a56f6b46c11d07b78262b4)

File: README.md (hunk 1 above)
Comment (ASendellPrice) [reply README.md]: Updated as requested. See commit [e6bc2b5](https://github.com/PatrickWeller/PanelPal/pull/74/commits/e6bc2b5a274e5a8801a56f6b46c11d07b78262b4)

File: test/test_check_panel.py
Comment (ASendellPrice) [reply test/test_check_panel.py:1]: Test coverage increased to 95% in commit [c94e5f2](https://github.com/PatrickWeller/PanelPal/pull/74/commits/c94e5f2017bacbaec2306799e7ab5e5034c4505d)
image

**Review by PatrickWeller — APPROVED** [review]
All looks good now Ash :1st_place_medal:  :)
